tests/topotests/lib/topotest.py

```diff
@@ -252,6 +252,54 @@ def run_and_expect(func, what, count=20, wait=3):
     return (False, result)
 
 
+def run_and_expect_type(func, etype, count=20, wait=3, avalue=None):
+    """
+    Run `func` and compare the result with `etype`. Do it for `count` times
+    waiting `wait` seconds between tries. By default it tries 20 times with
+    3 seconds delay between tries.
+
+    This function is used when you want to test the return type and,
+    optionally, the return value.
+
+    Returns (True, func-return) on success or
+    (False, func-return) on failure.
+    """
+    start_time = time.time()
+    func_name = "<unknown>"
+    if func.__class__ == functools.partial:
+        func_name = func.func.__name__
+    else:
+        func_name = func.__name__
```

Comment:
Callers passing partials should probably be calling [`update_wrapper`](https://docs.python.org/2/library/functools.html#functools.update_wrapper) rather than doing this type check here.

Reply:
There are a lot of callers using `partial` (check the simpler `run_and_expect` function). Beyond avoiding this `if` `else` here I don't see a gain using it everywhere. Feel free to do it or suggest a better way to use it, the smaller the idiom the better.

Please let me know if I'm not seeing something here.

Reply:
Sure, I don't care enough to argue about it